snapcraft/parts/lifecycle.py

```diff
@@ -74,13 +77,34 @@ def get_snap_project() -> _SnapProject:
     )
 
 
+def apply_yaml(yaml_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Apply Snapcraft logic to yaml_data."""
+    # validate project grammar
+    GrammarAwareProject.validate_grammar(yaml_data)
+
+    # Special Snapcraft Part
+    core_parts = {k: yaml_data.pop(k) for k in _CORE_PART_KEYS if k in yaml_data}
```

Comment:
Nice way to create the core part! I think it sould be called `core_part` since it's just one?